### PR TITLE
docs: add self signed certs guide

### LINKED ISSUE #13781: Node fails to join with self-signed certificates
_Umbrella: Improve user experience for users trying teleport with self-signed certificates._

The node can't join because of the x.509 cert connecting to proxy. Node has failed with confusing error message:

```
Failed to resolve tunnel address Get "https://teleport.domain.com:443/webapi/find": x509: certificate signed by unknown authority reversetunnel/transport.go:90
```

In this case, the error message should have been (with a proper guide linked)

```
Failed to connect to the target proxy. Error: x509: certificate signed by unknown authority. Your proxy certificate is not trusted or expired. Please update certificate or follow this guide for self-signed certs: https://goteleport.com/docs/getting-started/self-signed#node
```

Please write the guide and update the section with the systemd unit to have insecure flag if running on Linux, or with helm chart for agent.

Don't forget to add `sudo systemctl daemon-reload` and `systemctl restart teleport.service` in the guide.

### PR DESCRIPTION
**Pull request by GavinFrazar**
Closes #13781

This PR adds a guide for using self-signed certs with Teleport. It includes a prominent warning about why you should never use self-signed certs in production, and info about how to use `teleport`, `tctl`, and `tsh` with a proxy that uses self-signed certs.

For using `teleport` I included info about passing `--insecure` for helm and for systemd daemon.

The PR also consolidates copies of the same snippet regarding TLS certificate setup into a single include file: 
`tls-certificate-setup.mdx`

### CODE REVIEW

**Comment by GavinFrazar** [comment]
I also moved the info on how to setup certs into an include file and included that in the self-signed certs guide

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (ptgott) [inline docs/pages/management/admin/self-signed-certs.mdx:107]: I don't think we need this section, since this guide is explicitly intended for users who have already decided to use self-signed certificates in an insecure environment. If we do want to direct the user to instructions on setting up proper PKI with Teleport, we could, for example, link to https://goteleport.com/docs/deploy-a-cluster/open-source/

File: docs/pages/management/admin.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin.mdx]: 😅 Yes, it's hard to summarize without getting too verbose, but how is this: 
```diff
-Run Teleport with self-signed certs.
+Set up Teleport in a local environment without configuring TLS certificates.
```

I considered "dev environment" but idk if that works. Basically "an environment for trying out teleport that isn't production". Not sure.

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:7]: I included that in the warning <Notice> below but I could repeat it here?

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:9]: The proxy service runs the web UI and proxies API calls - in both cases it authenticates to users via x509 certs signed by a CA that the user's machine trust, like from LetsEncrypt for example. So you need the certs configured for secure use of the web UI or `tsh`

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:11]: All of these tools are used to interact with a Teleport cluster.
```suggestion
which clients will not trust by default. This can cause problems interacting with your Teleport cluster via Teleport cli tools, such as
```

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:19]: I'd like to keep the link to the proper configuration section below, but sure let's remove the wiki link to MITM attacks

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:7]: Nvm, I'll move it from the warning up to the intro where it makes more sense.

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:30]: Good point, and the Teleport Cloud tab doesnt make sense since in the context of this article either. I'll remove that include and make a custom one for this guide

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:63]: Sure, but for completion let's keep the link and add a nested tab for showing how to do it inline

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:81]: Why are we avoiding links in guides? Links are informative, concise, keep the guide from veering off-topic, and provide a single source of truth. If I inline an explanation for how to setup a systemd unit file, then that info has to be kept in sync with the systemd guide. It's also more information than I'd like to include in this guide. If I omit the link and don't inline, then the reader who doesn't know how to setup systemd has to go find that guide themselves.

We should be cautious of being overzealous in avoiding links. In this case, I've provided the additional info a user needs to modify their systemd unit file, assuming they know what a unit file is and how systemd services work. If they don't know then I provide a link to our systemd guide. I think that's worth including as a "see more info" link, because anyone who knows what I'm talking about already won't click the link, and anyone who doesn't know can click the link.

Having typed up my argument for links, maybe I should rephrase the sentence around links like this: `If you are unfamiliar with X, you can learn about it here: <link>`

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:87]: Sounds good

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:107]: ha, this is actually a case where I think inlining is warranted since the guide emphasizes the importance of setting up certs properly in prod. I don't feel strongly about it though, so I can replace the inline with a link here

**Comment by GavinFrazar** [comment]
@ptgott this is ready for re-review, could you take another look?

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (ptgott) [inline docs/pages/management/admin/self-signed-certs.mdx:187]: I think we can make "use: `tctl --insecure`" more precise. Right now, it's not clear if this is a full command or not. Maybe say, "use the `--insecure` flag when running `tctl` commands"?

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (ptgott) [inline docs/pages/management/admin/self-signed-certs.mdx:188]:
```suggestion

`tctl` will determine how to connect to the Auth Service in a few ways:
```
I've suggested capitalizing "Auth Service" and adding a new paragraph since this is the start of a new thought.

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (ptgott) [inline docs/pages/management/admin/self-signed-certs.mdx:189]:
```suggestion
- loading configuration from a local profile after logging in with `tsh`
```
Since we haven't mentioned "profile" yet, I've suggested adding an indefinite article and qualifying this a bit more.

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (ptgott) [reply docs/pages/management/admin/self-signed-certs.mdx]: I'm not sure this guide is about the importance of setting up certs properly. To me, up until this section, it's explicitly focused on how to use self-signed certificates in non-production environments. If we want to reframe this guide to be about how the Proxy Service uses TLS in general, we could include this section, but we should also:
- move this to a different section of the docs, e.g., `/docs/architecture`
- change the title and introductory section to be about how the Proxy Service uses TLS in general
- make this more of a conceptual guide than a how-to guide

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (ptgott) [reply docs/pages/management/admin/self-signed-certs.mdx:177]: The [style guide](https://goteleport.com/docs/contributing/documentation/style-guide/#expectations) advises against adding links to the body sections of how-to guides because the goal of these guides is to get the reader up and running with a particular setup as quickly as possible. 

If we want to include a link in a how-to guide, I think we should make it explicit what a reader should get out of the link, e.g., something concrete like a configuration value, JSON document to copy/paste, etc. If we're telling a reader to consult a link to learn more information, we're not providing them with a quick path toward completing a goal.

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (ptgott) [inline docs/pages/management/admin/self-signed-certs.mdx:165]: Maybe add an instruction to run `systemctl status teleport` to get the path to the unit file?

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:177]: > If we're telling a reader to consult a link to learn more information, we're not providing them with a quick path toward completing a goal.

I see, in that case I agree we should remove the link here. I can always put "further reading" links at the bottom of the page.

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx]: This is related to the link discussion we had above, and I think in this case I'll just remove the inline guide and provide a link under "Further Reading" at the bottom of the page.

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:17]: You pretty much hit the nail on the head that the problem in general is going to be connecting to the proxy service, so I'll change this to be more specific.

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:166]: Good idea, added

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx]: removing link to the config section below which described how to setup TLS certs, since I'm removing that section

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:188]: Changed according to your suggestion. I also added a "For example: `tctl status --insecure`" at the end of tctl section, and "For example: `tsh login --proxy=tele.example.com:443 --user=alice --insecure`" at the end of the tsh section.

**Comment by GavinFrazar** [comment]
@ptgott I incorporated your latest feedback, please take a look when you can. Sorry this is taking up so much of your time

**Review by ptgott — APPROVED** [review]
Thanks!

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (ryanclark) [inline docs/pages/management/admin/self-signed-certs.mdx:107]: We added config v3 which has either `auth_server` or `proxy_server` 

https://github.com/gravitational/teleport/blob/bd0ae3819b27e107153398de926392c96f8088ae/docs/pages/reference/config.mdx?plain=1#L142-L158

Can you update this to match please?

Note, we didn't backport this, so when backporting for v10 you should still mention `auth_servers`.

**Comment by alexfornuto** [comment]
I disabled auto-merge and reassigned back to @GavinFrazar to address the latest review.

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx:107]: Sure. On that note, are we planning to update the `teleport start` cli to have separate `--proxy-server` and `--auth-server`?

Comment (ryanclark) [reply docs/pages/management/admin/self-signed-certs.mdx]: cc @zmb3

**Comment by ryanclark** [comment]
@GavinFrazar did you mean to assign this PR to me?

File: docs/pages/management/admin/self-signed-certs.mdx
Comment (GavinFrazar) [reply docs/pages/management/admin/self-signed-certs.mdx]: thanks for catching this, I wasn't thinking 🥲

**Comment by GavinFrazar** [comment]
> @GavinFrazar did you mean to assign this PR to me?

I reassigned after addressing your feedback, since I've seen that a few times in other PRs I thought maybe it was just the norm lol.

**Comment by GavinFrazar** [comment]
@ryanclark just waiting on your approval, was there something else you wanted addressed still?

**Review by ryanclark — APPROVED** [review]

**Pull request merged**
